### DISCUSSION TITLE
Cannot get WordPress debug.log working.

### DISCUSSION THREAD

**Discussion by ChrisTaylorDeveloper** [docker-library/wordpress#890] [General]
I cannot get WordPress in this Docker image to create a `debug.log` file.  I am using the Docker Compose example from here https://hub.docker.com/_/wordpress

I set the `WORDPRESS_DEBUG` environment variable, in the wordpress service definition and then call the `error_log('...') `function from a plugin.  No `debug.log` file appears in the `wp-content` folder or anywhere else that I can find.

Any advice please?

**Comment by tianon** [comment]
I think if you want `debug.log`, you might _also_ need to add `define( 'WP_DEBUG_LOG', true );` to `wp-config` as noted in https://developer.wordpress.org/advanced-administration/debug/debug-wordpress/ (most easily via `WORDPRESS_CONFIG_EXTRA` :eyes:)

**Comment by yosifkit** [comment]
Unless you are overriding `debug_log` in the php configuration, then it will not be in a file and just show up in `docker logs`:

https://github.com/docker-library/wordpress/blob/654752dda3941244b10ab45a07bf693d39a05d7c/Dockerfile.template#L168

**Comment by ChrisTaylorDeveloper** [comment]
Let me check if I can see my log messages in docker logs.  Thank you.

**Comment by ChrisTaylorDeveloper** [comment]
My log messages do appear in the docker logs.  And they appear even when I do not set `WORDPRESS_DEBUG`.
So, that begs the question, what is the function of WORDPRESS_DEBUG?  Is there some functionality missing in the Image?  Happy to open an Issue if required.

**Reply by tianon** [reply]
To be clear, you see _your_ `error_log` invocations with or without it, right?  I think that's because `error_log` is a PHP construct, not a WordPress specific one: https://www.php.net/manual/en/function.error-log.php

(and `WORDPRESS_DEBUG` controls WordPress behavior itself, as described in https://developer.wordpress.org/advanced-administration/debug/debug-wordpress/#wp_debug)

**Comment by ChrisTaylorDeveloper** [comment]
Yes, I see the result of `error_log()` whether or not I start the container with `WORDPRESS_DEBUG`.

To get `/wp-content/debug.log` working, I think it would be reasonable to start the container with both `WORDPRESS_DEBUG` and (a new long-form) `WORDPRESS_DEBUG_LOG`.  Or, pass `WP_DEBUG_LOG` via `WORDPRESS_CONFIG_EXTRA`.  Either way, the Image would need to be adjusted to stop using stderr I guess?

**Reply by tianon** [reply]
WordPress itself adjusts the PHP configuration automatically based on `WP_DEBUG_LOG`: https://github.com/WordPress/WordPress/blob/1dc3f878703ff788a5dc1b06efda6c665554dcb3/wp-includes/load.php#L593-L596

We do not plan to add more environment variables, so `WORDPRESS_CONFIG_EXTRA` is going to remain our official answer for how to use this upstream WordPress functionality.

**Comment by ChrisTaylorDeveloper** [comment]
I have this working now.  Thank you.
